### ISSUE TITLE
environment

### ISSUE THREAD

**Issue by jikim-research** [gaobb/AnoGen#10]
Hello, I used python 3.10.14 with the following package versions for DIFFUSION.

(AnoGen) root@3321da4eb464:/fake_defect_generator# pip list
Package                     Version
--------------------------- -----------
absl-py                     2.1.0
academictorrents            2.3.3
aiofiles                    23.2.1
aiohappyeyeballs            2.4.0
aiohttp                     3.10.5
aiosignal                   1.3.1
albucore                    0.0.17
albumentations              1.4.16
annotated-types             0.7.0
anomalib                    1.1.1
antlr4-python3-runtime      4.9.3
anyio                       4.6.2.post1
async-timeout               4.0.3
attrs                       24.2.0
bencode.py                  2.0.0
bitstring                   3.1.5
certifi                     2024.8.30
charset-normalizer          3.3.2
click                       8.1.7
clip                        0.2.0
contourpy                   1.3.0
cycler                      0.12.1
docstring_parser            0.16
einops                      0.8.0
eval_type_backport          0.2.0
evaluation                  0.0.2
exceptiongroup              1.2.2
fastapi                     0.115.4
ffmpy                       0.4.0
filelock                    3.16.1
fonttools                   4.54.0
frozenlist                  1.4.1
fsspec                      2024.9.0
future                      0.16.0
glog                        0.3.1
gradio                      5.4.0
gradio_client               1.4.2
grpcio                      1.66.1
h11                         0.14.0
httpcore                    1.0.6
httpx                       0.27.2
huggingface-hub             0.25.1
idna                        3.10
imageio                     2.35.1
imgaug                      0.4.0
importlib_resources         6.4.5
jedi                        0.19.1
Jinja2                      3.1.4
joblib                      1.4.2
jsonargparse                4.32.1
kiwisolver                  1.4.7
kornia                      0.6.12
lazy_loader                 0.4
Markdown                    3.7
markdown-it-py              3.0.0
MarkupSafe                  2.1.5
matplotlib                  3.9.2
mdurl                       0.1.2
multidict                   6.1.0
natsort                     8.4.0
networkx                    3.3
numpy                       1.24.4
nvidia-cublas-cu11          11.10.3.66
nvidia-cuda-nvrtc-cu11      11.7.99
nvidia-cuda-runtime-cu11    11.7.99
nvidia-cudnn-cu11           8.5.0.96
omegaconf                   2.3.0
opencv-python               4.10.0.84
opencv-python-headless      4.10.0.84
orjson                      3.10.11
packaging                   24.1
pandas                      2.2.3
parso                       0.8.4
pillow                      10.4.0
pip                         24.2
protobuf                    3.20.1
pudb                        2024.1.2
pydantic                    2.9.2
pydantic_core               2.23.4
pyDeprecate                 0.3.2
pydub                       0.25.1
Pygments                    2.18.0
pyparsing                   3.1.4
PyPubSub                    3.3.0
python-dateutil             2.9.0.post0
python-gflags               3.1.2
python-multipart            0.0.12
pytorch-lightning           1.6.5
pytz                        2024.2
PyYAML                      6.0.2
regex                       2024.9.11
requests                    2.32.3
rich                        13.8.1
rich-argparse               1.5.2
ruff                        0.7.2
safehttpx                   0.1.1
safetensors                 0.4.5
scikit-image                0.24.0
scikit-learn                1.5.2
scipy                       1.14.1
semantic-version            2.10.0
setuptools                  75.1.0
shapely                     2.0.6
shellingham                 1.5.4
six                         1.16.0
sniffio                     1.3.1
starlette                   0.41.2
taming-transformers-rom1504 0.0.6
tensorboard                 2.17.1
tensorboard-data-server     0.7.2
tensorboardX                2.6.2.2
test_tube                   0.7.5
threadpoolctl               3.5.0
tifffile                    2024.9.20
timm                        1.0.9
tokenizers                  0.13.3
tomlkit                     0.12.0
torch                       1.13.1
torchmetrics                0.11.4
torchvision                 0.14.1
tqdm                        4.64.1
transformers                4.28.1
typer                       0.12.5
typeshed_client             2.7.0
typing_extensions           4.12.2
tzdata                      2024.2
urllib3                     2.2.3
urwid                       2.6.15
urwid_readline              0.15.1
uvicorn                     0.32.0
wcwidth                     0.2.13
websockets                  12.0
Werkzeug                    3.0.4
wheel                       0.44.0
yarl                        1.12.1

I've tried to train defect embedding for mvtec dataset and generated defects in the bounding box, but the results are inconsistent with those reported in the paper.

![image](https://github.com/user-attachments/assets/38774e4e-dd9b-4394-ac4f-181aab4c118c)
![image](https://github.com/user-attachments/assets/6eea8793-2056-4cd9-9812-54c4157c2fed)
![image](https://github.com/user-attachments/assets/ef724e7a-4420-4082-9cc9-ae189d9fad03)

Could you please specify the exact python version, cuda version, library versions you used in the DIFFUSION  for reproducibility?
Thank you!!😊
